modules/calib3d/src/rhorefc.cpp

```diff
@@ -0,0 +1,2426 @@
+ *                         inliers for acceptance was reached; 0 otherwise.
+ */
+
+unsigned rhoRefC(RHO_HEST_REFC* p,       /* Homography estimation context. */
+                 const float*   src,     /* Source points */
+                 const float*   dst,     /* Destination points */
+                 char*          inl,     /* Inlier mask */
+                 unsigned       N,       /*  = src.length = dst.length = inl.length */
+                 float          maxD,    /* Works:     3.0 */
+                 unsigned       maxI,    /* Works:    2000 */
+                 unsigned       rConvg,  /* Works:    2000 */
+                 double         cfd,     /* Works:   0.995 */
+                 unsigned       minInl,  /* Minimum:     4 */
+                 double         beta,    /* Works:    0.35 */
+                 unsigned       flags,   /* Works:       0 */
+                 const float*   guessH,  /* Extrinsic guess, NULL if none provided */
+                 float*         finalH){ /* Final result. */
+    return p->rhoRefC(src, dst, inl, N, maxD, maxI, rConvg, cfd, minInl, beta,
+                      flags, guessH, finalH);
+}
+
+
+
+/**
+ * Allocate memory aligned to a boundary of MEMALIGN.
+ */
+
+static inline void*  almalloc(size_t nBytes){
+    if(nBytes){
+        unsigned char* ptr = (unsigned char*)malloc(MEM_ALIGN + nBytes);
+        if(ptr){
+            unsigned char* adj = (unsigned char*)(((intptr_t)(ptr+MEM_ALIGN))&((intptr_t)(-MEM_ALIGN)));
+            ptrdiff_t diff = adj - ptr;
+            adj[-1] = (unsigned char)(diff - 1);
+            return adj;
+        }
+    }
+
+    return NULL;
+}
+
+/**
+ * Free aligned memory.
+ *
+ * If argument is NULL, do nothing in accordance with free() semantics.
+ */
+
+static inline void   alfree(void* ptr){
+    if(ptr){
+        unsigned char* cptr = (unsigned char*)ptr;
+        free(cptr - (ptrdiff_t)cptr[-1] - 1);
+    }
+}
+
+/**
+ * Constructor for RHO_HEST_REFC.
+ *
+ * Does nothing. True initialization is done by initialize().
+ */
+
+RHO_HEST_REFC::RHO_HEST_REFC() : init(0){
+
+}
+
+/**
+ * Private copy constructor for RHO_HEST_REFC. Disabled.
+ */
+
+RHO_HEST_REFC::RHO_HEST_REFC(const RHO_HEST_REFC&) : init(0){
+
+}
+
+/**
+ * Destructor for RHO_HEST_REFC.
+ */
+
+RHO_HEST_REFC::~RHO_HEST_REFC(){
+    if(init){
+        finalize();
+    }
+}
+
+
+
+/**
+ * Initialize the estimator context, by allocating the aligned buffers
+ * internally needed.
+ *
+ * Currently there are 5 per-estimator buffers:
+ * - The buffer of m indexes representing a sample
+ * - The buffer of 16 floats representing m matches (x,y) -> (X,Y).
+ * - The buffer for the current homography
+ * - The buffer for the best-so-far homography
+ * - Optionally, the non-randomness criterion table
+ *
+ * Returns 0 if unsuccessful and non-0 otherwise.
+ */
+
+inline int    RHO_HEST_REFC::initialize(void){
+    ctrl.smpl   = (unsigned*)almalloc(SMPL_SIZE*sizeof(*ctrl.smpl));
```

Comment:
please, switch completely to std::vector, cv::Mat, cv::Matx for arrays. cv::Ptr&lt;&gt; for pointers to classes/structures